packs/BP/scripts/network.ts

```diff
@@ -89,123 +93,152 @@ export class MachineNetwork extends DestroyableObject {
    * automatically sets each generator's storage to the amount it sent that was not received.
    * returns automatically if the object is not valid.
    */
-  private *send(): Generator<void, void, void> {
+  private *send(): Generator<undefined, void, unknown> {
     if (!this.isValid) return;
 
-    interface Target {
-      block: Block;
-      amount: number;
-      definition: InternalRegisteredMachine;
-    }
+    // Calculate the amount of each type that is avaliable to send around.
+    const distribution: Record<string, number> = {};
 
-    const targets: Record<string, Target[]> = {};
+    this.sendQueue.forEach((send) => {
+      const newValue = distribution[send.type] ?? 0;
+      distribution[send.type] = newValue + send.amount;
+    });
 
-    while (this.sendQueue.length) {
-      const queuedSend = this.sendQueue.pop()!;
-      if (!(queuedSend.type in targets)) {
-        const targetsArr: Target[] = [];
+    // Clear queue for next time.
+    this.sendQueue = [];
+    const typesToDistribute = Object.keys(distribution);
 
-        for (const block of this.connections.machines) {
-          if (
-            !block.hasTag(
-              `fluffyalien_energisticscore:consumer.${queuedSend.type}`,
-            ) &&
-            !block.hasTag("fluffyalien_energisticscore:consumer._any")
-          ) {
-            yield;
-            continue;
-          }
+    interface ConsumerGroups {
+      normalPriority: Block[];
+      lowPriority: Block[];
+    }
 
-          const definition = machineRegistry[block.typeId] as
-            | InternalRegisteredMachine
-            | undefined;
+    // initialize consumers keys.
+    const consumers: Record<string, ConsumerGroups> = {};
+    for (const key of typesToDistribute) {
+      consumers[key] = { lowPriority: [], normalPriority: [] };
+    }
 
-          if (!definition) {
-            logWarn(
-              `can't send '${queuedSend.type}' to machine '${block.typeId}': this block is configured as a machine but it couldn't be found in the machine registry`,
-            );
-            yield;
-            continue;
-          }
+    // find and filter connections into their consumer groups.
+    this.connections.machines.forEach((machine) => {
+      const machineTags = machine.getTags();
+      const isLowPriority = machineTags.includes(
+        "fluffyalien_energisticscore:low_priority_consumer",
+      );
+      const allowsAny = machineTags.includes(
+        "fluffyalien_energisticscore:consumer._any",
+      );
 
-          const amount = getMachineStorage(block, queuedSend.type);
+      // Check machine tags and sort into appropriate groups.
+      Object.keys(consumers).forEach((consumerType) => {
```

Comment:
use `typesToDistribute` instead of `Object.keys(consumers)`, it is both clearer and avoids an extra `Object.keys` call